apis/compute/v1alpha1/computeregiontargettcpproxy_types.go

```diff
@@ -0,0 +1,122 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	refs "github.com/GoogleCloudPlatform/k8s-config-connector/apis/refs/v1beta1"
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/apis/k8s/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var ComputeRegionTargetTCPProxyGVK = GroupVersion.WithKind("ComputeRegionTargetTCPProxy")
+
+// ComputeRegionTargetTCPProxySpec defines the desired state of ComputeRegionTargetTCPProxy
+// +kcc:proto=google.cloud.compute.v1.TargetTcpProxy
+type ComputeRegionTargetTCPProxySpec struct {
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="BackendServiceRef is immutable"
+	// Immutable. A reference to the ComputeBackendService resource.
+	// +required
+	BackendServiceRef *refs.ComputeBackendServiceRef `json:"backendServiceRef"`
+
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="Description is immutable"
+	// Immutable. An optional description of this resource.
+	Description *string `json:"description,omitempty"`
```

Comment:
oh, it's interesting that the description is immutable.

Reply:
Actually regional TCP proxy is immutable, there's no "update" function. https://cloud.google.com/compute/docs/reference/rest/v1/regionTargetTcpProxies